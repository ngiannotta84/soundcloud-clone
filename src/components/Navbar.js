import React from "react";
import { Link } from "react-router-dom";
import "../styles/navbar.css";

const Navbar = () => {
  return (
    <div className="navbar">
      <ul className="navbar-links">
        <li>
          <Link className="navbar-links-item" to="/">
            Home
          </Link>
        </li>
        <li>
          <Link className="navbar-links-item" to="login">
            Login
          </Link>
        </li>
        <li>
          <Link className="navbar-links-item" to="profile">
            Profile
          </Link>
        </li>
        <li>
          <Link className="navbar-links-item" to="search">
            Search
          </Link>
        </li>
        <li>
          <Link className="navbar-links-item" to="upload">
            Upload
          </Link>
        </li>
      </ul>
    </div>
  );
};
export default Navbar;
