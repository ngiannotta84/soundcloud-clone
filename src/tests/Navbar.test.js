import React from "react";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Navbar from "../components/Navbar";

describe("Navbar", () => {
  const validProps = {
    userName: "validName",
  };
  const mockNav = jest.fn();
  jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockNav,
  }));

  describe("snapshots", () => {
    test("userName is falsey snapshot", () => {
      const { asFragment } = render(
        <Router>
          <Navbar />
        </Router>
      );

      expect(asFragment()).toMatchSnapshot();
    });

    test("userName is truthy snapshot", () => {
      const { asFragment } = render(
        <Router>
          <Navbar userName={validProps.userName} />
        </Router>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("userName is falsey tests", () => {
    beforeEach(() => {
      render(
        <Router>
          <Navbar />
        </Router>
      );
    });

    test("renders correctly", () => {
      expect(screen.getByText(/soundclone/i).closest("a")).toHaveAttribute(
        "href",
        "/"
      );
      expect(screen.getByText(/home/i).closest("a")).toHaveAttribute(
        "href",
        "/"
      );
      expect(screen.getByText(/login/i).closest("a")).toHaveAttribute(
        "href",
        "/login"
      );
      expect(screen.getByAltText(/home/i)).toHaveAttribute("src", "home.png");
      expect(screen.getByAltText(/search/i)).toHaveAttribute(
        "src",
        "search.png"
      );
      expect(screen.getByAltText(/login/i)).toHaveAttribute("src", "login.png");
    });

    test("search bar", () => {
      const searchInput = screen.getByRole("textbox");
      const searchButton = screen.getByAltText(/search/i).closest("button");
      const searchText = "text";

      fireEvent.change(searchInput, { target: { value: searchText } });
      fireEvent.click(searchButton);

      waitFor(() => {
        expect(mockNav).toBeCalledWith(`search/${searchText}`);
      });
    });
  });

  describe("userName is truthy tests", () => {
    beforeEach(() => {
      render(
        <Router>
          <Navbar userName={validProps.userName} />
        </Router>
      );
    });

    test("renders links", () => {
      expect(screen.getByText(/soundclone/i).closest("a")).toHaveAttribute(
        "href",
        "/"
      );
      expect(screen.getByText(/home/i).closest("a")).toHaveAttribute(
        "href",
        "/"
      );
      expect(screen.getByText(/profile/i).closest("a")).toHaveAttribute(
        "href",
        `/profile/${validProps.userName}`
      );
      expect(screen.getByText(/upload/i).closest("a")).toHaveAttribute(
        "href",
        "/upload"
      );
      expect(screen.getByText(/logout/i).closest("a")).toHaveAttribute(
        "href",
        "/logout"
      );
      expect(screen.getByAltText(/home/i)).toHaveAttribute("src", "home.png");
      expect(screen.getByAltText(/search/i)).toHaveAttribute(
        "src",
        "search.png"
      );
      expect(screen.getByAltText(/profile/i)).toHaveAttribute(
        "src",
        "profile.png"
      );
      expect(screen.getByAltText(/upload/i)).toHaveAttribute(
        "src",
        "upload.png"
      );
      expect(screen.getByAltText(/logout/i)).toHaveAttribute(
        "src",
        "logout.png"
      );
    });
  });
});
