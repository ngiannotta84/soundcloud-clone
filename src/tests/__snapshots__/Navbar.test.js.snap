// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshots userName is falsey snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <ul
      class="navbar__links"
    >
      <li
        class="navbar__links__item"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <h1
            class="navbar__links__item__heading"
          >
            SoundClone
          </h1>
        </a>
      </li>
      <li
        class="navbar__links__item"
      >
        <a
          aria-current="page"
          class="navbar__links__item--active"
          href="/"
        >
          <img
            alt="home"
            class="navbar__links__item__img"
            src="home.png"
          />
          <span
            class="navbar__links__item__text"
          >
            home
          </span>
        </a>
      </li>
      <li
        class="navbar__links__search"
      >
        <input
          type="text"
        />
        <button
          type="button"
        >
          <img
            alt="search"
            class="navbar__links__search__img"
            src="search.png"
          />
        </button>
      </li>
      <li
        class="navbar__links__item"
      >
        <a
          href="/login"
        >
          <img
            alt="login"
            class="navbar__links__item__img"
            src="login.png"
          />
          <span
            class="navbar__links__item__text"
          >
            login
          </span>
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Navbar snapshots userName is truthy snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <ul
      class="navbar__links"
    >
      <li
        class="navbar__links__item"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <h1
            class="navbar__links__item__heading"
          >
            SoundClone
          </h1>
        </a>
      </li>
      <li
        class="navbar__links__item"
      >
        <a
          aria-current="page"
          class="navbar__links__item--active"
          href="/"
        >
          <img
            alt="home"
            class="navbar__links__item__img"
            src="home.png"
          />
          <span
            class="navbar__links__item__text"
          >
            home
          </span>
        </a>
      </li>
      <li
        class="navbar__links__search"
      >
        <input
          type="text"
        />
        <button
          type="button"
        >
          <img
            alt="search"
            class="navbar__links__search__img"
            src="search.png"
          />
        </button>
      </li>
      <li
        class="navbar__links__item"
      >
        <a
          href="/profile/validName"
        >
          <img
            alt="profile"
            class="navbar__links__item__img"
            src="profile.png"
          />
          <span
            class="navbar__links__item__text"
          >
            profile
          </span>
        </a>
      </li>
      <li
        class="navbar__links__item"
      >
        <a
          href="/upload"
        >
          <img
            alt="upload"
            class="navbar__links__item__img"
            src="upload.png"
          />
          <span
            class="navbar__links__item__text"
          >
            upload
          </span>
        </a>
      </li>
      <li
        class="navbar__links__item"
      >
        <a
          href="/logout"
        >
          <img
            alt="logout"
            class="navbar__links__item__img"
            src="logout.png"
          />
          <span
            class="navbar__links__item__text"
          >
            logout
          </span>
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
